In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [5]:
import os
os.listdir()

['.ipynb_checkpoints',
 'gender_submission.csv',
 'test.csv',
 'train.csv',
 'Untitled.ipynb']

In [3]:
data1=pd.read_csv("gender_submission.csv")
data2=pd.read_csv("test.csv")
data3=pd.read_csv("train.csv")
data3.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [4]:
#children_age<18 
#adult_age >=18
#male with children
#children with parents
#with sibling
#with spouse

[m,n]=data3.shape
temp_gender={'male':[0,0],'female':[0,0]}#classify either nake or female survival

temp_age={"female":{"withSibSp":{"adult":[0,0],"child":[0,0]},"withoutSibSp":{"adult":[0,0],"child":[0,0]},"withParch":{"adult":[0,0],"child":[0,0]},"withoutParch":{"adult":[0,0],"child":[0,0]}},"male":{"withSibSp":{"adult":[0,0],"child":[0,0]},"withoutSibSp":{"adult":[0,0],"child":[0,0]},"withParch":{"adult":[0,0],"child":[0,0]},"withoutParch":{"adult":[0,0],"child":[0,0]}}}
#classify baed on age and gender

for i in range(m):
    temp_gender[data3['Sex'][i]][data3['Survived'][i]]+=1
    rnge=''
    if data3["Age"][i]>=18:#18 is the age when I assume to be the starting point of adulthood
        rnge="adult"
    else:
        rnge="child"
    sibSp=""
    if data3["SibSp"][i]!=0:
        sibSp="withSibSp"
    else:
        sibSp="withoutSibSp"
    
    parch=""
    if data3["Parch"][i]!=0:
        parch="withParch"
    else:
        parch="withoutParch"
        
    temp_age[data3['Sex'][i]][parch][rnge][data3['Survived'][i]]+=1
    temp_age[data3['Sex'][i]][sibSp][rnge][data3['Survived'][i]]+=1

    
print(temp_gender,"\n\n")
print(temp_age)




    

    

{'male': [468, 109], 'female': [81, 233]} 


{'female': {'withSibSp': {'adult': [22, 64], 'child': [22, 32]}, 'withoutSibSp': {'adult': [25, 95], 'child': [12, 42]}, 'withParch': {'adult': [17, 53], 'child': [23, 27]}, 'withoutParch': {'adult': [30, 106], 'child': [11, 47]}}, 'male': {'withSibSp': {'adult': [69, 18], 'child': [38, 18]}, 'withoutSibSp': {'adult': [256, 52], 'child': [105, 21]}, 'withParch': {'adult': [36, 6], 'child': [28, 23]}, 'withoutParch': {'adult': [289, 64], 'child': [115, 16]}}}


In [5]:
#calcualtion for evaluation

#total no of female
total_female=int(sum(temp_gender['female']))
#total no of survived female
survived_female=temp_gender['female'][1]
#total no of male
total_male=sum(temp_gender['male'])
#total no of male survive
survived_male=temp_gender['male'][1]
#percentage of female survive
percentage_female_survived=(survived_female/total_female)*100
#percentage of male survive
percentage_male_survived=(survived_male/total_male)*100
#percentage of female in survival people
percentage_female_from_survival=(survived_female/(survived_female+survived_male))*100
#percentage of people survive 
percentage_people_Survived=((survived_male+survived_female)/(total_female+total_male))*100



print(percentage_female_survived,percentage_male_survived,percentage_female_from_survival,percentage_people_Survived)
print("\n total no of passenger in titanic=\t",total_female+total_male,
     "\n \n columns>>\n total_count>> total male or female", "\n survived_count>> total no of female|male survived","\npercentage_of_total>> percentage among all the population"," \npercentage_survived>>> percentage of male|female survived among own gender ","\n percentage_among_survived>>> percentage of male|female among all the survived people"," \nsurvived_percentage_of_total>>>survived among all the population")


temp_dict={'gender':['male','female'],'total_count':[total_male,total_female],"survived_count":[survived_male,survived_female],'percentage_survived':[percentage_male_survived,percentage_female_survived]
         ,'survived_percent_of_total':[(survived_male/(total_male+total_female))*100,(survived_female/(total_male+total_female))*100] 
          ,"percentage_of_total":[100*(total_male/891),100*(total_female/891)]
           ,"percentage_among_survived":[100-percentage_female_from_survival,percentage_female_from_survival]
          }
df=pd.DataFrame.from_dict(temp_dict)

df

74.20382165605095 18.890814558058924 68.12865497076024 38.38383838383838

 total no of passenger in titanic=	 891 
 
 columns>>
 total_count>> total male or female 
 survived_count>> total no of female|male survived 
percentage_of_total>> percentage among all the population  
percentage_survived>>> percentage of male|female survived among own gender  
 percentage_among_survived>>> percentage of male|female among all the survived people  
survived_percentage_of_total>>>survived among all the population


gender  percentage_among_survived  percentage_of_total  \
0    male                  31.871345            64.758698   
1  female                  68.128655            35.241302   

   percentage_survived  survived_count  survived_percent_of_total  total_count  
0            18.890815             109                  12.233446          577  
1            74.203822             233                  26.150393          314

# Inside Information

* 18.89% of male on other hand 74.20% female survived . So 55.31 % more female survive than male .
* among the survive people 31.87 % percentage is male and 68.13 % are female

In [5]:

df_age_dict={"class":["Total","not survive","survive","% survival","% survival_with_Parrents/child","% survival_without_Parrents/child","% survival_with_siblings/spouse","% survival_without_siblings/spouse"]}
for x in temp_age.keys():
    #adults
    
    #total no of adults
    total=int(sum(temp_age[x]["withParch"]["adult"])+sum(temp_age[x]["withoutParch"]["adult"]))
    #total of adults not survived
    notsurvive=temp_age[x]["withParch"]["adult"][0]+temp_age[x]["withoutParch"]["adult"][0]
    #total no of adults who survived
    survive=temp_age[x]["withParch"]["adult"][1]+temp_age[x]["withoutParch"]["adult"][1]
    #percentage of adults survive
    percentageSurvive=100*survive/total
    #percentage of survival who are with parents/chirldren
    percentage_survival_Parch=100*temp_age[x]["withParch"]["adult"][1]/total
    #percentage of  survival who are without parents/chirldren
    percentage_survival_withoutParch=100*temp_age[x]["withoutParch"]["adult"][1]/total
    #percentage of survival who are with siblings/spouse
    percentage_survival_SibSp=100*temp_age[x]["withSibSp"]["adult"][1]/total
    #percentage of  survival who are without siblings/spouse
    percentage_survival_withoutSibSp=100*temp_age[x]["withoutSibSp"]["adult"][1]/total
    
    #create the dictionary 
    temp_list=[total,notsurvive,survive,percentageSurvive,percentage_survival_Parch,percentage_survival_withoutParch,percentage_survival_SibSp,percentage_survival_withoutSibSp]
    temp_list= [ '%.2f' % elem for elem in temp_list ]
    df_age_dict[str(x)+"_adult"]=temp_list
    
    
    #child
    #total no of childs
    total=sum(temp_age[x]["withParch"]["child"])+sum(temp_age[x]["withoutParch"]["child"])
    #total no of childs not survive
    notsurvive=temp_age[x]["withParch"]["child"][0]+temp_age[x]["withoutParch"]["child"][0]
    #total no of childs  survive
    survive=temp_age[x]["withParch"]["child"][1]+temp_age[x]["withoutParch"]["child"][1]
    #percentage of child survival
    percentageSurvive=100*survive/total
    #percentage of survival who are with parents/chirldren
    percentage_survival_Parch=100*(temp_age[x]["withParch"]["child"][1]/total)
    #percentage of survival who are without parents/chirldren
    percentage_survival_withoutParch=100*(temp_age[x]["withoutParch"]["child"][1]/total)
    #percentage of survival who are with siblings/spouse
    percentage_survival_SibSp=100*(temp_age[x]["withSibSp"]["child"][1]/total)
    #percentage of survival who are without siblings/spouse
    percentage_survival_withoutSibSp=100*(temp_age[x]["withoutSibSp"]["child"][1]/total)
    
    
    temp_list=[total,notsurvive,survive,percentageSurvive,percentage_survival_Parch,percentage_survival_withoutParch,percentage_survival_SibSp,percentage_survival_withoutSibSp]
    temp_list= [ '%.2f' % elem for elem in temp_list ]
    
    df_age_dict[str(x)+"_child"]=temp_list
    #for y in temp_age[x]:
        #df_age_dict[str(x)+"_"+str(y)]=[sum(temp_age[x][y])]+temp_age[x][y]+[100*(temp_age[x][y][1])/sum(temp_age[x][y])]
df_age=pd.DataFrame(df_age_dict)
df_age

class female_adult female_child male_adult  \
0                               Total       206.00       108.00     395.00   
1                         not survive        47.00        34.00     325.00   
2                             survive       159.00        74.00      70.00   
3                          % survival        77.18        68.52      17.72   
4      % survival_with_Parrents/child        25.73        25.00       1.52   
5   % survival_without_Parrents/child        51.46        43.52      16.20   
6     % survival_with_siblings/spouse        31.07        29.63       4.56   
7  % survival_without_siblings/spouse        46.12        38.89      13.16   

  male_child  
0     182.00  
1     143.00  
2      39.00  
3      21.43  
4      12.64  
5       8.79  
6       9.89  
7      11.54

# Information

### if we classify age>=18 as adults and under 18 as child

* then most probabilty is with the female adults(77.18%) and least is with male adults(17.72%)

* if the passenger is with Parents or child then  the passenger has maximum probabilty of survival if the passnger is female adult(25.72%)

* if the passenger is without Parents or child then  the passenger has maximum probabilty of survival if the passnger is female adult(51.45%)

* males_adults passenger are seems to be more careful to their loved one as whenever they have someone with them their probabilty of surviving decrease drustically



In [6]:
[m,n]=data3.shape
temp_pclass={1:{'male':[0,0],'female':[0,0],"count":0},2:{'male':[0,0],'female':[0,0],"count":0},3:{'male':[0,0],'female':[0,0],"count":0}}
for i in range(m):
    #in which class who survive either male or female
    temp_pclass[data3['Pclass'][i]][data3['Sex'][i]][data3['Survived'][i]]+=1
    #no of survival in class
    temp_pclass[data3['Pclass'][i]]["count"]+=1
temp_pclass

{1: {'count': 216, 'female': [3, 91], 'male': [77, 45]},
 2: {'count': 184, 'female': [6, 70], 'male': [91, 17]},
 3: {'count': 491, 'female': [72, 72], 'male': [300, 47]}}

In [7]:
temp_dictClass={'Class':["1st","2nd","3rd"],"No_of_traveller":[temp_pclass[1]['count'],temp_pclass[2]['count'],temp_pclass[3]['count']],
               "Total_survived":[temp_pclass[1]['male'][1]+temp_pclass[1]['female'][1],temp_pclass[2]['male'][1]+temp_pclass[2]['female'][1],temp_pclass[3]['male'][1]+temp_pclass[3]['female'][1]  ]
               ,"Total_male_survive":[temp_pclass[1]['male'][1],temp_pclass[2]['male'][1],temp_pclass[3]['male'][1]]
                ,"Total_female_survive":[temp_pclass[1]['female'][1],temp_pclass[2]['female'][1],temp_pclass[3]['female'][1]]
                ,"Percentage_male_survive":[(temp_pclass[1]['male'][1]/sum(temp_pclass[1]['male']))*100,(temp_pclass[2]['male'][1]/sum(temp_pclass[2]['male']))*100,(temp_pclass[3]['male'][1]/sum(temp_pclass[3]['male']))*100  ]
               ,"Percentage_female_survive":[(temp_pclass[1]['female'][1]/sum(temp_pclass[1]['female']))*100,(temp_pclass[2]['female'][1]/sum(temp_pclass[2]['female']))*100,(temp_pclass[3]['female'][1]/sum(temp_pclass[3]['female']))*100  ]
                ,"Percentage_survive":[((temp_pclass[1]['female'][1]+temp_pclass[1]['male'][1])/temp_pclass[1]['count'])*100,((temp_pclass[2]['female'][1]+temp_pclass[2]['male'][1])/temp_pclass[2]['count'])*100,((temp_pclass[3]['female'][1]+temp_pclass[3]['male'][1])/temp_pclass[3]['count'])*100]
               }
df_class=pd.DataFrame(temp_dictClass)
df_class

Class  No_of_traveller  Percentage_female_survive  Percentage_male_survive  \
0   1st              216                  96.808511                36.885246   
1   2nd              184                  92.105263                15.740741   
2   3rd              491                  50.000000                13.544669   

   Percentage_survive  Total_female_survive  Total_male_survive  \
0           62.962963                    91                  45   
1           47.282609                    70                  17   
2           24.236253                    72                  47   

   Total_survived  
0             136  
1              87  
2             119

# Information

* percentage of survival is more in 1st class tickets than the other classes and the survival percentages are different for male and female... In 1st class 62.96% people, in 2nd class 47.28% and in 3rd class 24.23% of people survive.

* There are more people travelling in 3rd class than the 1st and 2nd class 

* females have more propability of survival than male passenger in any class of reservation.

In [8]:
#depending upon the passenger Id
def prime(num):
    if num > 1:
        for i in range(2,int(num/2)+1):
            if (num % i) == 0:
                return 0
        return 1
       
# if input number is less than
# or equal to 1, it is not prime
    else:
       return 0


temp_passId={2:{"count":0,'male':[0,0],'female':[0,0]},3:{"count":0,'male':[0,0],'female':[0,0]},5:{"count":0,'male':[0,0],'female':[0,0]},10:{"count":0,'male':[0,0],'female':[0,0]},"prime":{"count":0,'male':[0,0],'female':[0,0]}}

[m,n]=data3.shape

for i in range(m):
    #if passenger Id is divisible by 2
    if data3["PassengerId"][i]%2==0:
        temp_passId[2]["count"]+=1
        temp_passId[2][data3['Sex'][i]][data3['Survived'][i]]+=1
    
    #if passenger Id is divisible by 3
    if data3["PassengerId"][i]%3==0:
        temp_passId[3]["count"]+=1
        temp_passId[3][data3['Sex'][i]][data3['Survived'][i]]+=1
        
    #if passenger Id is divisible by 5
    if data3["PassengerId"][i]%5==0:
        temp_passId[5]["count"]+=1
        temp_passId[5][data3['Sex'][i]][data3['Survived'][i]]+=1
    
    #if passenger Id is divisible by 10
    if data3["PassengerId"][i]%10==0:
        temp_passId[10]["count"]+=1
        temp_passId[10][data3['Sex'][i]][data3['Survived'][i]]+=1
    
    #if passengerId is a prime no
    if prime(data3["PassengerId"][i]):
        temp_passId["prime"]["count"]+=1
        temp_passId["prime"][data3['Sex'][i]][data3['Survived'][i]]+=1
        
temp_passId     



{10: {'count': 89, 'female': [8, 31], 'male': [36, 14]},
 2: {'count': 445, 'female': [36, 121], 'male': [225, 63]},
 3: {'count': 297, 'female': [26, 68], 'male': [167, 36]},
 5: {'count': 178, 'female': [21, 44], 'male': [88, 25]},
 'prime': {'count': 154, 'female': [14, 50], 'male': [77, 13]}}

In [11]:
temp_passDict={"TYPE":["div by 2","div by 3","div by 5","div by 10","prime no"],"total":[temp_passId[x]["count"] for x in [2,3,5,10,"prime"] ]
              ,"total_survive":[temp_passId[x]["male"][1]+temp_passId[x]["female"][1] for x in [2,3,5,10,"prime"]]
               ,"total_survive%":[100*(temp_passId[x]["male"][1]+temp_passId[x]["female"][1])/temp_passId[x]["count"] for x in [2,3,5,10,"prime"]]
               ,"no_of_female":[sum(temp_passId[x]["female"]) for x in [2,3,5,10,"prime"]],"no_of_male":[sum(temp_passId[x]["male"]) for x in [2,3,5,10,"prime"]]
               ,"no_female_survive":[temp_passId[x]["female"][1] for x in [2,3,5,10,"prime"]],"no_male_survive":[temp_passId[x]["male"][1] for x in [2,3,5,10,"prime"]]
               ,"no_female_survive%":[100*(temp_passId[x]["female"][1])/sum(temp_passId[x]["female"]) for x in [2,3,5,10,"prime"]],"no_male_survive%":[100*(temp_passId[x]["male"][1])/sum(temp_passId[x]["male"]) for x in [2,3,5,10,"prime"]]
              }

df_pass=pd.DataFrame(temp_passDict)
df_pass

TYPE  no_female_survive  no_female_survive%  no_male_survive  \
0   div by 2                121           77.070064               63   
1   div by 3                 68           72.340426               36   
2   div by 5                 44           67.692308               25   
3  div by 10                 31           79.487179               14   
4   prime no                 50           78.125000               13   

   no_male_survive%  no_of_female  no_of_male  total  total_survive  \
0         21.875000           157         288    445            184   
1         17.733990            94         203    297            104   
2         22.123894            65         113    178             69   
3         28.000000            39          50     89             45   
4         14.444444            64          90    154             63   

   total_survive%  
0       41.348315  
1       35.016835  
2       38.764045  
3       50.561798  
4       40.909091

# information

* the passenger having passengerID divisible by 10 has most probality of survival which is around 50.56% where females have 79.49% and males have 28.0% of survival .

* females have most probabilty of survival in the passengerId divisible by 10 followed by the IDs which are prime no on the other hand males have most probability on the passengerID which are divisible by 10 followed by the Ids which are divisible by 5.